web/client/utils/styleparser/StyleParserUtils.js

```diff
@@ -917,7 +918,10 @@ export const drawIcons = (geoStylerStyle, options) => {
     }, []);
     const marks = symbolizers.filter(({ kind }) => kind === 'Mark');
     const icons = symbolizers.filter(({ kind }) => kind === 'Icon');
-    return loadFontAwesome()
+    const loadFontAwesomeForIcons = getConfigProp("loadFontAwesomeForIcons");
+    // if undefined or true it will load it to preserve previous behaviour
+    const loadingPromise =  (isNil(loadFontAwesomeForIcons) || loadFontAwesomeForIcons) && icons?.length ? loadFontAwesome() : Promise.resolve();
+    return loadingPromise
```

Comment:
@allyoucanmap 
I think the use case we opened this issue is in importing vector layer. In import new vector layer -- > **drawIcons** method is called and it calls **loadFontAwesome** within it by default. We don't make exporting this specific method and use it particularly in import. 
So I think your approach doesn't prevent calling this method because in import vector layer, calling this method is not optional.